core/src/main/java/google/registry/flows/domain/DomainCheckFlow.java

```diff
@@ -163,7 +163,8 @@ public EppResponse run() throws EppException {
                     clientId,
                     now));
 
-    ImmutableList.Builder<DomainCheck> checks = new ImmutableList.Builder<>();
+    ImmutableList.Builder<DomainCheck> checksBuilder = new ImmutableList.Builder<>();
+    ImmutableSet.Builder<String> availableDomains = new ImmutableSet.Builder<>();
```

Comment:
I tried several different approaches here before I settled on this one.  A Map of domain names to Checks doesn't work because there can be duplicate domains in the check.  A Multimap doesn't work because it no longer preserves strict insertion order (multiple values for the same key end up collated next to each other), while the checks themselves must be in strict order because they aren't associated with a domain name except by index.

So this ends up being the simplest thing that works.